yamux/src/connection/stream.rs

```diff
@@ -429,30 +452,90 @@ impl AsyncWrite for Stream {
         self.sender
             .start_send(cmd)
             .map_err(|_| self.write_zero_err())?;
-        self.shared()
-            .update_state(self.conn, self.id, State::SendClosed);
+        self.shared.with_mut(|inner| {
+            inner.update_state(self.conn, self.id, State::SendClosed);
+        });
```

Comment:
Same here: i think we can use directly `Shared::update_state`?

Reply:
Done.